dev/app-shell/elements/persistent-arc.html

```diff
@@ -16,16 +16,32 @@
         db: db.child('arcs')
       };
     }
+    _getAdditionalRecipe() {
+      const params = (new URL(document.location)).searchParams;
+      // Prioritize manifest over solo, semi-arbitrarily, since usually we'll
+      // only see one or the other.
+      return params.get('solo') || params.get('manifest');
+    }
     _update(props, state, lastProps) {
       if (props.key === '*' && lastProps.key != props.key) {
         this._createKey(state.db);
       }
       if (props.key && props.key !== '*') {
-        if (props.metadata && props.metadata !== state.metadata) {
-          state.metadata = props.metadata;
-          let arcMetadata = state.db.child(props.key).child('metadata');
-          PersistentArc.log('WRITING (update) metadata for', String(arcMetadata), props.metadata);
-          arcMetadata.update(props.metadata);
+        if (props.metadata) {
+          // Typical developer workflow involves creating a new arc and
+          // subsequently modifying the url to include a specific recipe via a
+          // solo or manifest query param, thus we have to look for such a param
+          // at arc update time. Alternatively we could have the developer
+          // include the param in the main launcher page and have the app shell
+          // pass it along to the 'New Arc' url, but that is not the current
+          // state of the world.
+          props.metadata['additionalRecipe'] = this._getAdditionalRecipe();
```

Comment:
It feels a bit hacky to do this on every update, open to other thoughts and see comment above. I also considered checking if the value is undefined and if so deleting the `additionalRecipe` key but that seems overkill unless a reviewer thinks we do care.

Reply:
One question that we didn't fully resolve, and we probably won't know until later is if we should allow further updates the development makes directly to the URL to be reflected in this persistent state.

It may be really useful to allow a developer to change the manifest easily via the URL and to see that reflected in the persistent store, since that matches their intent.

There's an argument to be made the other way (that updating the URL shouldn't trigger a persistent change), too, that that @shaper and I discussed. As I type this, though, I'm not sure I can make that argument as well. Changing the manifest on the fly will change the course of the arc and reverting back to the old recipe isn't guaranteed to work either. So maybe it's best to just press on?

So this makes sense to me until we have our versioning story locked down.